### PR TITLE
StdlibUnittest: port the test harness to Windows

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift [hunk 1]
```diff
@@ -749,12 +756,23 @@ func _stdlib_getline() -> String? {
 func _printDebuggingAdvice(_ fullTestName: String) {
   print("To debug, run:")
   var invocation = [CommandLine.arguments[0]]
+#if os(Windows)
+  var buffer: UnsafeMutablePointer<CChar>?
+  var length: Int = 0
+  if _dupenv_s(&buffer, &length, "SWIFT_INTERPRETER") != 0 {
+    if buffer != nil {
```
Comment (troughton) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift]: Very minor drive-by code-style: this might be cleaner as `if let buffer = buffer`. The two `if` conditions could also be folded together:

```
if _dupenv_s(&buffer, &length, "SWIFT_INTERPRETER") != 0, let buffer = buffer {
```

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @xwu @moiseev

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @moiseev

**Comment by compnerd** [comment]
CC: @lorentey

**Review by moiseev — APPROVED** [review]
LGTM
